test_rmw_implementation/test/test_subscription.cpp

```diff
@@ -377,3 +377,92 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, publisher_count);
 }
+
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_sequence) {
+  rmw_ret_t ret;
+  size_t count = 1u;
+  size_t taken;
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  rmw_message_sequence_t sequence = rmw_get_zero_initialized_message_sequence();
+  ret = rmw_message_sequence_init(&sequence, count, &allocator);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  rmw_message_info_sequence_t info_sequence = rmw_get_zero_initialized_message_info_sequence();
+  ret = rmw_message_info_sequence_init(&info_sequence, count, &allocator);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  rmw_subscription_allocation_t * null_allocation{nullptr}; // still valid allocator
+
+  ret = rmw_take_sequence(sub, count, &sequence, &info_sequence, &taken, null_allocation);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+
+  ret = rmw_message_sequence_fini(&sequence);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+
+  ret = rmw_message_info_sequence_fini(&info_sequence);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+}
+
+
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_sequence_with_bad_args) {
+  rmw_ret_t ret;
```

Comment:
Same comment about init and declaring `ret` in the same place

Reply:
Updated on dc8ab77